File: api/javer/src/main/java/com/uuhnaut69/javer/model/Restaurant.java
package com.uuhnaut69.javer.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.OneToMany;
import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "restaurants")
public class Restaurant extends BaseEntity {

  private String name;

  private String address;

  @OneToMany(mappedBy = "restaurant", orphanRemoval = true, cascade = CascadeType.ALL)
  private Set<Dish> dishes = new HashSet<>();
}

File: kover/src/main/kotlin/com/uuhnaut69/kover/model/Restaurant.kt
package com.uuhnaut69.kover.model

import javax.persistence.CascadeType
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.OneToMany

@Entity(name = "restaurants")
class Restaurant : BaseEntity() {

    @Column(columnDefinition = "text", nullable = false)
    var name: String? = null

    @Column(columnDefinition = "text", nullable = false)
    var address: String? = null

    @OneToMany(mappedBy = "restaurant", orphanRemoval = true, cascade = [CascadeType.ALL])
    var dishes: Set<Dish>? = null
}